[Feat] DiaryInput 컴포넌트

Sungwoo00 opened the pull request:
## PR 유형
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR 체크리스트
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

### PR 상세

- 타입값 textArea 또는 input 선택에 따른 조건부 렌더링
- placeholder 제거 및 label로 대체
- input, textarea를 label 과 연결
- tm함수 적용
- input, textarea에 값 입력 시 label값 숨김 처리

Sungwoo00 committed:
[Feat] DiaryInput 컴포넌트

- 타입값 textArea 또는 input 선택에 따른 조건부 렌더링

MinQyu reviewed (changes requested):
다이어리 본문 부분은 분기 렌더링 대신 css로 구현해주세요!

Sungwoo00 commented:
> 다이어리 본문 부분은 분기 렌더링 대신 css로 구현해주세요!

현재 input 컴포넌트 하나에서 css만 변경 되도록 구현 하긴 했는데 생각해보니 줄바꿈 같은 기능이 필요할 거 같다는 생각이 들었습니다.
의견 공유 부탁드리겠습니다.

Sungwoo00 committed:
[Refactor] Diary컴포넌트

- type값에 따른 input, textarea 조건부 렌더링
- placeholder 제거 및 label로 대체
- input, textarea를 label 과 연결
- tm함수 적용
- input, textarea에 값 입력 시 label값 숨김 처리

MinQyu reviewed (approved):
고생하셨습니다~

BomEllen reviewed (approved):
수고하셨습니다!
